src/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,245 @@
+
+#include "data/DBHelpers.h"
+#include "rpc/AMMHelpers.h"
+
+#include <ripple/protocol/AMMCore.h>
+
+namespace {
+
+std::string
+toIso8601(ripple::NetClock::time_point tp)
+{
+    using namespace std::chrono;
+    static auto constexpr rippleEpochOffset = seconds{rippleEpochStart};
+
+    return date::format(
+        "%Y-%Om-%dT%H:%M:%OS%z",
+        date::sys_time<system_clock::duration>(system_clock::time_point{tp.time_since_epoch() + rippleEpochOffset})
+    );
+};
+
+}  // namespace
+
+namespace rpc {
+AMMInfoHandler::Result
+AMMInfoHandler::process(AMMInfoHandler::Input input, Context const& ctx) const
+{
+    using namespace ripple;
+
+    auto const hasInvalidParams = [&input] {
+        // no asset/asset2 can be specified if amm account is specified
+        if (input.ammAccount)
+            return input.issue1 != ripple::noIssue() || input.issue2 != ripple::noIssue();
+
+        // both assets must be specified when amm account is not specified
+        return input.issue1 == ripple::noIssue() || input.issue2 == ripple::noIssue();
+    }();
+
+    if (hasInvalidParams)
+        return Error{Status{RippledError::rpcINVALID_PARAMS}};
+
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_, ctx.yield, input.ledgerHash, input.ledgerIndex, range->maxSequence
+    );
+
+    if (auto const status = std::get_if<Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<LedgerInfo>(lgrInfoOrStatus);
+
+    if (input.accountID) {
+        auto keylet = keylet::account(*input.accountID);
+        if (not sharedPtrBackend_->fetchLedgerObject(keylet.key, lgrInfo.seq, ctx.yield))
+            return Error{Status{RippledError::rpcACT_NOT_FOUND}};
+    }
+
+    ripple::uint256 ammID;
+    if (input.ammAccount) {
+        auto const accountKeylet = keylet::account(*input.ammAccount);
+        auto const accountLedgerObject =
+            sharedPtrBackend_->fetchLedgerObject(accountKeylet.key, lgrInfo.seq, ctx.yield);
+        if (not accountLedgerObject)
+            return Error{Status{RippledError::rpcACT_MALFORMED}};
+        ripple::STLedgerEntry const sle{
+            ripple::SerialIter{accountLedgerObject->data(), accountLedgerObject->size()}, accountKeylet.key
+        };
+        if (not sle.isFieldPresent(ripple::sfAMMID))
+            return Error{Status{RippledError::rpcACT_NOT_FOUND}};
+        ammID = sle.getFieldH256(ripple::sfAMMID);
+    }
+
+    auto ammKeylet = ammID != 0 ? keylet::amm(ammID) : keylet::amm(input.issue1, input.issue2);
+    auto const ammBlob = sharedPtrBackend_->fetchLedgerObject(ammKeylet.key, lgrInfo.seq, ctx.yield);
+
+    if (not ammBlob)
+        return Error{Status{RippledError::rpcACT_NOT_FOUND}};
+
+    auto const amm = SLE{SerialIter{ammBlob->data(), ammBlob->size()}, ammKeylet.key};
+    auto const ammAccountID = amm.getAccountID(sfAccount);
+    auto const accBlob =
```

Comment:
We can skip this check if input.ammAccount has value. input.ammAccount should be the same with the sfAccount in amm object.

Reply:
That may be the case but I'd rather keep the logic/code as close to `rippled` as possible. Let's not add extras for now.